tests/tests.js

```diff
@@ -358,44 +380,68 @@ exports.defineAutoTests = function() {
-                        expect(obj.note).toBe(noteText);
-                        expect(obj.birthday.toDateString()).toBe(bDay.toDateString());
-                        expect(obj.emails.length).toBe(1);
-                        expect(obj.emails[0].value).toBe('here@there.com');
-                        removeContact(); // Clean up contact object
-                        done();
+                var obj = {
+                    "gender": "male",
+                    "note": "my note",
+                    "name": {
+                        "familyName": "Delete",
+                        "givenName": "Test"
                     },
-                    fail = function() {
-                        removeContact();
-                        fail(done);
-                    };
+                    "emails": [{
+                        "value": "here@there.com"
+                    }, {
+                        "value": "there@here.com"
+                    }],
+                    "birthday": aDay
+                };
+
+                var saveFail = fail.bind(null, done);
+
+                var saveSuccess = function(obj) {
+                    // must store returned object in order to have id for update test below
+                    gContactObj = obj;
+                    gContactObj.emails[1].value = "";
+                    bDay = new Date(1975, 5, 4);
+                    gContactObj.birthday = bDay;
+                    gContactObj.note = noteText;
+                    gContactObj.save(updateSuccess, saveFail);
+                };
+
+                var updateSuccess = function(obj) {
+                    expect(obj).toBeDefined();
+                    expect(obj.id).toBe(gContactObj.id);
+                    expect(obj.note).toBe(noteText);
+                    expect(obj.birthday.toDateString()).toBe(bDay.toDateString());
+                    expect(obj.emails.length).toBe(1);
+                    expect(obj.emails[0].value).toBe('here@there.com');
+                    done();
+                };
+
+                navigator.contacts
+                    .create(obj)
+                    .save(saveSuccess, saveFail);
 
-                // remove an email
-                gContactObj.emails[1].value = "";
-                // change birthday
-                gContactObj.birthday = bDay;
-                // update note
-                gContactObj.note = noteText;
-                gContactObj.save(win, fail);
             }, MEDIUM_TIMEOUT);
         });
+
         describe('Contact.remove method', function(done) {
-            afterEach(removeContact);
+            afterEach(function (done) {
+                removeContact(done);
+            });
 
             it("contacts.spec.22 calling remove on a contact has an id of null should return ContactError.UNKNOWN_ERROR", function(done) {
-                var win = function() {};
+                var win = function() {
+                    expect(false).toBe(true);
```

Comment:
IMO this might be rewritten to use the global `fail` function as in spec.23 and others, i.e. use `fail.bind` as it does the same thing.